tests/qasm/test_open_qasm_2_and_3.py

```diff
@@ -50,6 +51,37 @@ def test_circular_dependency_detection_false_positive():
         assert False, f"Circular dependency raised while it shouldn't"
 
 
+def test_in_time_gate_def_3_to_2():
+    converted_file_name = qasm_folder + "in_time_gate_def.qasm"
+    file_name = qasm_folder + "in_time_gate_def_converted.qasm"
+    with open(converted_file_name, "r") as f:
+        assert ' '.join(
+            open_qasm_file_conversion_3_to_2(file_name).split()
+        ) == ' '.join(f.read().split())
+
+
+def test_late_gate_def_3_to_2():
+    with pytest.raises(ValueError):
+        open_qasm_file_conversion_3_to_2(qasm_folder + "late_gate_def.qasm")
+
+
+def test_circular_dependency_detection_3_to_2():
+    with pytest.raises(RuntimeError) as e:
+        open_qasm_file_conversion_3_to_2(
+            qasm_folder + "circular_dep1.qasm",
+        )
+    assert "Circular dependency" in str(e.value)
+
+
+def test_circular_dependency_detection_false_positive_3_to_2():
+    try:
+        open_qasm_file_conversion_3_to_2(
+            qasm_folder + "circular_dep_a_3.qasm",
+        )
+    except RuntimeError:
```

Comment:
you sure you need to catch the error here ? If the error is not supposed to happen, I would just write
```py
def test_circular_dependency_detection_false_positive_3_to_2():
    open_qasm_file_conversion_3_to_2(qasm_folder + "circular_dep_a_3.qasm")
```

Reply:
done